apps/test/unit/lib/util/timeoutApiTest.js

```diff
@@ -0,0 +1,330 @@
+
+      // Check executors map arguments to object correctly
+      let spy = sinon.spy();
+      injectExecuteCmd(spy);
+      executors[funcName]('one', 'two', 'three');
+      expect(spy).to.have.been.calledOnce;
+      expect(spy.firstCall.args[2]).to.deep.equal({ms: 'one', callback: 'two'});
+    });
+
+    itComplainsIfArgumentIsNotAFunction(funcName, 'callback', {
+      callback: () => {},
+      ms: 0,
+    });
+
+    itComplainsIfArgumentIsNotANumber(funcName, 'ms', {
+      callback: () => {},
+      ms: 0,
+    });
+  });
+
+  describe('stopTimedLoop', () => {
+    const funcName = 'stopTimedLoop';
+
+    it('has no default arguments, max one argument', () => {
+      // Check droplet config
+      expect(dropletConfig[funcName].paletteParams).to.be.undefined;
+      expect(dropletConfig[funcName].params).to.be.undefined;
+      expect(dropletConfig[funcName].paramButtons).to.deep.equal({minArgs: 0, maxArgs: 1});
+
+      // Check executors map arguments to object correctly
+      let spy = sinon.spy();
+      injectExecuteCmd(spy);
+      executors[funcName](); // Called with no args
+      expect(spy).to.have.been.calledOnce;
+      expect(spy.firstCall.args[2]).to.deep.equal({key: undefined});
+
+      spy = sinon.spy();
+      injectExecuteCmd(spy);
+      executors[funcName]('one', 'two'); // Called with extra args
+      expect(spy).to.have.been.calledOnce;
+      expect(spy.firstCall.args[2]).to.deep.equal({key: 'one'});
+    });
+
+    itComplainsIfArgumentIsNotANumber(funcName, 'key', {
+      key: 0,
+    });
+  });
+
+  function itComplainsIfArgumentIsNotANumber(funcName, argName, validArgs) {
+    it(`complains if argument ${argName} is not a number`, () => {
+      commands[funcName]({
+        ...validArgs,
+        [argName]: null
+      });
+      expect(testErrorHandler.outputWarning).to.have.been.calledOnce;
+      expect(testErrorHandler.outputWarning.firstCall.args[0]).to.equal(
+        `${funcName}() ${argName} parameter value (null) is not a number.`
+      );
+
+      commands[funcName]({
+        ...validArgs,
+        [argName]: 'string'
+      });
+      expect(testErrorHandler.outputWarning).to.have.been.calledTwice;
+      expect(testErrorHandler.outputWarning.secondCall.args[0]).to.equal(
+        `${funcName}() ${argName} parameter value (string) is not a number.`
+      );
+
+      commands[funcName]({
+        ...validArgs,
+        [argName]: {obj: 5}
+      });
+      expect(testErrorHandler.outputWarning).to.have.been.calledThrice;
+      expect(testErrorHandler.outputWarning.thirdCall.args[0]).to.equal(
+        `${funcName}() ${argName} parameter value ([object Object]) is not a number.`
+      );
+    });
+  }
+
+  function itComplainsIfArgumentIsNotAFunction(funcName, argName, validArgs) {
+    it(`complains if argument ${argName} is not a function`, () => {
+      commands[funcName]({
+        ...validArgs,
+        [argName]: null
```

Comment:
Very nice abstraction! It took me a while to wrap my head around it, and specifically why `funcName` isn't the only argument -- please add a function-level comment briefly saying what each argument does, which I think would suffice in explaining why so much stuff needs to be passed into this function.